### ISSUE TITLE
Add on-screen keys for special characters

### ISSUE THREAD

**Issue by schmittlauch** [kantord/LibreLingo#66]
While going through the "animals" example course, I discovered that when *león* was asked the answer *leon* (without accent) is still accepted as correct.

Is this intended behaviour or are you normalising the string to heavily before comparison? IMHO learning where to set accents in words is an important part of learning a language.

Apart from that, keep up the nice work.

**Comment by ryliejamesthomas** [comment]
For me it marks it as correct, but acknowledges that I left out the accent.

> Correct solution!
>
> You made a small error. Correct spelling: león

**Comment by Deuchnord** [comment]
While I agree that accents are important when learning languages, they become a problem when you are using a keyboard that makes them inaccessible.

For instance: I'm using a [French keyboard](https://commons.wikimedia.org/wiki/File:KB_France.svg), which gives you (logically) an advantage to use accents in French (éèêëàôöùûü), but (still logically) does not provide an easy way to type the Spanish-specific ones, nor the German-specific ß letter (for example). Linux users are a little advantaged here because they can configure the Compose key on their keyboard to fix most of these problem (e.g. ó = Compose + `'` + `o`), but AFAIK Windows and Mac users are still wronged.

IMO it might be great to add buttons in the interface to append those accents in the field.

**Comment by igalic** [comment]
this is all considerably easier on a mobile platform where *adding* a new keyboard layout and *seeing* what keys you're typing on is all very easy

this is me assuming you **can** see.
i don't know which accessibility affordances android and iOS have for the actual keyboard.

**Comment by kantord** [comment]
> Is this intended behaviour or are you normalising the string to heavily before comparison?

There is some normalization going on for answers, yes. The reason why it's accepted isn't the normalization though. Rather, a certain edit distance is allowed. See: https://github.com/kantord/LibreLingo/blob/master/src/components/ShortInputChallenge.svelte#L28

Small typos (and possibly even accent mark and punctuation mistakes) should be accepted, because it would be annoying to have to try again many times for tiny mistakes.

In the case of such typos, the answer is accepted, but the user is still alerted of the error, see:

![Screenshot_2020-02-09 LibreLingo - learn animals in Spanish for free](https://user-images.githubusercontent.com/3704904/74106787-cb23ed00-4b69-11ea-82ee-06a5f88ba1dc.png)

I think that's a good solution, but maybe the design is not good here? Maybe it could call for the user's attention a bit more.

**Comment by kantord** [comment]
> IMO it might be great to add buttons in the interface to append those accents in the field.

YES

**Comment by kantord** [comment]
@all-contributors please add @schmittlauch for ideas

**Comment by kantord** [comment]
@all-contributors please add @Deuchnord for ideas

**Comment by schmittlauch** [comment]
@kantord 
> maybe the design is not good here? Maybe it could call for the user's attention a bit more.

Definitely, the current design is so subtle that I didn't notice at all, being just a small line of text at the bottom edge of a maximised window on a large screen.
I suggest using a separate design and not treat them as correct but corretctly-ish. So let them fall in a different category then correct or incorrect and maybe represent them in yellow. It should be clearly noticeable that the answer was not totally correct.

**Comment by pinkprius** [comment]
It's this way because that's how Duolingo does it (I think). 
I think it works fine this way.

**Comment by schmittlauch** [comment]
@pinkprius You already have to know *that* you are looking for a small correction notification, unless you won't notice it.
![Screenshot_20200210_164959](https://user-images.githubusercontent.com/1479555/74165623-ad1bc280-4c25-11ea-926a-4445a3c2af31.png)

**Comment by pinkprius** [comment]
I agree, but I don't think that's necessarily bad. Maybe a system where you have stars depending how correct your answer is would make it clear?

**Comment by Deuchnord** [comment]
The alert is great, but not really adapted to wide screens. It might be better to move it closer to the form on these devices to make it more visible and thus, more accessible, mightn't it?

**Comment by kantord** [comment]
> this is me assuming you can see.
i don't know which accessibility affordances android and iOS have for the actual keyboard.

it's indeed an important point. Accessibility is very important. I hope someone with knowledge about it will take interest in contributing. I'll create a label for issues that are concerning accessibility

**Comment by kantord** [comment]
I created a new issue regarding the layout issue on wide screens: https://github.com/kantord/LibreLingo/issues/81

cc. @Deuchnord

**Comment by kantord** [comment]
@pinkprius 

> I agree, but I don't think that's necessarily bad. Maybe a system where you have stars depending how correct your answer is would make it clear?

I don't think there would be many different levels of correctness, although a complicated system might be devised for letting the user know about different errors.

Still, differentiating between a wholly correct error and an almost correct makes sense.

Making the color yellow might makes sense, as @schmittlauch says, but I think a slightly milder solution would be preferred

**Comment by kantord** [comment]
Solved in https://github.com/kantord/LibreLingo/pull/167
